src/WireFormat.h

```diff
@@ -1605,6 +1648,182 @@ struct TakeIndexletOwnership {
+                : type(OpType::READ)
+                , tableId(tableId)
+                , rpcId(rpcId)
+                , keyLength(keyLength)
+                , rejectRules(rejectRules)
+            {
+            }
+        } __attribute__((packed));
+
+        struct RemoveOp {
+            OpType type;
+            uint64_t tableId;
+            uint64_t rpcId;
+            uint16_t keyLength;
+            RejectRules rejectRules;
+
+            // In buffer: The actual key for this part
+            // follows immediately after this.
+            RemoveOp(uint64_t tableId, uint64_t rpcId, uint16_t keyLength,
+                       RejectRules rejectRules)
+                : type(OpType::REMOVE)
+                , tableId(tableId)
+                , rpcId(rpcId)
+                , keyLength(keyLength)
+                , rejectRules(rejectRules)
+            {
+            }
+        } __attribute__((packed));
+
+        struct WriteOp {
+            OpType type;
+            uint64_t tableId;
+            uint64_t rpcId;
+            uint32_t length;        // length of keysAndValue
+            RejectRules rejectRules;
+
+            // In buffer: KeysAndValue follow immediately after this
+            WriteOp(uint64_t tableId, uint64_t rpcId, uint32_t length,
+                        RejectRules rejectRules)
+                : type(OpType::WRITE)
+                , tableId(tableId)
+                , rpcId(rpcId)
+                , length(length)
+                , rejectRules(rejectRules)
+            {
+            }
+        } __attribute__((packed));
+    } __attribute__((packed));
+
+    struct Response {
+        ResponseCommon common;
+        Vote vote;
+    } __attribute__((packed));
+};
+
+struct TxRequestAbort {
+    static const Opcode opcode = Opcode::TX_REQUEST_ABORT;
+    static const ServiceType service = MASTER_SERVICE;
+
+    enum Vote { COMMIT, ABORT, INVALID };
+
+    struct Request {
+        RequestCommon common;
+        uint64_t leaseId; //Recovery coordinator may not know about leaseTerm.
+                          //DM can set arbitrary leaseTerm anyway.
+        uint32_t participantCount; // Number of local objects participating TX
+                                   // for this server.
+        // List of local participants.
+
+    } __attribute__((packed));
+
+    struct Response {
+        ResponseCommon common;
+        Vote vote;
+    } __attribute__((packed));
+};
+
+struct TxHintFailed {
+    static const Opcode opcode = Opcode::TX_HINT_FAILED;
+    static const ServiceType service = MASTER_SERVICE;
+
+    struct Request {
+        RequestCommon common;
+        uint64_t leaseId;           // Id of the client lease associated with
+                                    // this transaction.
+        uint32_t participantCount;  // Number of local objects participating TX
+                                    // for this server.
+        // List of local Participants
```

Comment:
This is a list of *all* participants, not just local ones, right?  And, it has a different structure than for a local participant, right? Give the particular structure type.